homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -0,0 +1,73 @@
+"""Suppoort for Amcrest IP camera binary sensors."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.const import CONF_NAME, CONF_BINARY_SENSORS
+from . import DATA_AMCREST, BINARY_SENSORS
+
+DEPENDENCIES = ['amcrest']
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=5)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up a binary sensor for an Amcrest IP Camera."""
+    if discovery_info is None:
+        return
+
+    device_name = discovery_info[CONF_NAME]
+    binary_sensors = discovery_info[CONF_BINARY_SENSORS]
+    amcrest = hass.data[DATA_AMCREST][device_name]
+
+    amcrest_binary_sensors = []
+    for sensor_type in binary_sensors:
+        amcrest_binary_sensors.append(
+            AmcrestBinarySensor(amcrest.name, amcrest.device, sensor_type))
+
+    async_add_devices(amcrest_binary_sensors, True)
+    return True
+
+
+class AmcrestBinarySensor(BinarySensorDevice):
+    """Binary sensor for Amcrest camera."""
+
+    def __init__(self, name, camera, sensor_type):
+        """Initialize entity."""
+        self._name = '{} {}'.format(name, BINARY_SENSORS[sensor_type])
+        self._camera = camera
+        self._sensor_type = sensor_type
+        self._state = None
+
+    @property
+    def name(self):
+        """Return entity name."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return if entity is on."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return device class."""
+        return 'motion'
```

Comment:
Do we have const for this?

Reply:
There wasn't when I wrote this, and when I searched for one just a short while ago. However, looks like constants for the binary_sensor device classes were just added 2 days ago by balloob. I can rebase and use it.